fix: correct signature of gvRenderData() function

mcepl opened the pull request:
For compatibility with graphviz 14.*

Yes, it adds `swig` as build time dependency, because I think it is The Right Thing™ to do to use the original source. However, it can be changed into using generated C file (which should be then properly packaged).

Fixes: https://github.com/pygraphviz/pygraphviz/issues/567
Replaces: https://github.com/pygraphviz/pygraphviz/pull/566

mcepl committed:
fix: correct signature of gvRenderData() function

For compatibility with graphviz 14.*

Fixes: https://github.com/pygraphviz/pygraphviz/issues/567
Replaces: https://github.com/pygraphviz/pygraphviz/pull/566
Signed-off-by: Matěj Cepl <mcepl@cepl.eu>

Smattr commented:
This feels like the right direction to me. Though unfortunately the break occurred in 13.0.0 and the `GRAPHVIZ_MAJOR_VERSION` macro was only introduced in 14.0.0. If you want more robust behaviour across [13.0.0, 14.0.0) and can rely on the binaries being installed, you could try something like:

```python
match = re.match(
  r"dot - graphviz version (\d+)",
  subprocess.check_output(["dot", "-V"], stderr=subprocess.STDOUT)
)
```

rossbar committed:
MAINT: Add swig step to build.

Rm autogenerated graphviz_wrap.c file, which is a product of
swig conversion of graphviz.i.

Co-authored-by: Matěj Cepl <mcepl@cepl.eu>

rossbar commented:
Okay - as mentioned in #567 , the graphviz dependency problem is fixed (or will be after release) for folks installing from wheels.

For folks installing from source however, we still have this problem. Having thought about this a bit, 2 approaches occur to me:

1. Dynamic version-detection at build-time; i.e. something akin to what's proposed here.
2. A lower-bound pin on graphviz versions

Which one makes the most sense depends on the packagers' needs. IMO option 2 is preferable for pygraphviz - it obviates the need for version-based switches in the code and adding infrastructure to test against different graphviz versions. However, if there is a strong need to support many different versions of graphviz, then that changes the picture. Thoughts?

rossbar commented:
Thanks again @mcepl and @Smattr - the core here were incorporated in #603 (with attribution) so I'll go ahead and close this one!
